src/Components/SmartComponents/SystemDetail/SystemDetail.js

```diff
@@ -33,7 +32,7 @@ class SystemDetail extends React.Component {
         });
 
         if (!entity) {
-            return <InvalidObject />;
+            return null;
```

Comment:
since inventory handles the empty state, we don't have to check it and return `null`, right? 
Adding `                    {entity && <Vulnerabilities entity={entity.id} />}` will handle the undefined entity in case it's not found or in invalid